plugin/src/main/java/org/jenkinsci/plugins/workflow/cps/CpsFlowExecution.java

```diff
@@ -840,51 +839,17 @@ public void onFailure(Throwable t) {
     }
 
     /**
-     * Used by {@link #loadProgramAsync(File)} to propagate a failure to load the persisted execution state.
-     * <p>
-     * Let the workflow interrupt by throwing an exception that indicates how it failed.
+     * Used to propagate a failure to load the persisted execution state.
      * @param promise same as {@link #programPromise} but more strongly typed
      */
     private void loadProgramFailed(final Throwable problem, SettableFuture<CpsThreadGroup> promise) {
-        FlowHead head;
-
-        synchronized(this) {
-            if (heads == null || heads.isEmpty()) {
```

Comment:
I do wonder if these cases were significant and ignoring them could cause problems in `onProgramEnd` in `croak`. Given they apparently have no specific test coverage though it does seem simplest to just ignore them for now.

Reply:
Maybe? I am not sure what purpose is served by making a fake head. AFAICT the `head` calculation here was in order to attach the _Failed to load build state_ error, but this does not appear to work anyway.